modules/programs/wezterm.nix

```diff
@@ -0,0 +1,595 @@
+    ];
+
+  toBindingMods = binding:
+    optionalString (binding.modifiers != [ ])
+    ''mods = "${concatStringsSep "|" binding.modifiers}",'';
+
+  toBindingAction = binding:
+    optionalString (binding.action != "") "action = ${binding.action},";
+
+  toWeztermBindings = bindings: isKeyBinding:
+    let
+      name = if isKeyBinding then "keys" else "mouse_bindings";
+      comment_name = if isKeyBinding then "Key Bindings" else "Mouse Bindings";
+      midBinding = binding:
+        if isKeyBinding then
+          ''key = "${binding.key}",''
+        else ''
+          event = {
+              ${binding.event} = {
+                streak = ${toString binding.count},
+                button = "${binding.button}",
+              },
+            },'';
+      mapBinding = binding: ''
+        {
+          ${toBindingMods binding}
+          ${midBinding binding}
+          ${toBindingAction binding}
+        },'';
+    in ''
+      -- ${comment_name}
+        ${name} = {
+          ${formatBindings (map mapBinding bindings)}
+        },'';
+
+  toWeztermTabColors = tab: indent:
+    let
+      formatted = ''
+        {
+          bg_color = "${tab.background}",
+          fg_color = "${tab.foreground}",
+          intensity = "${tab.intensity}",
+          italic = ${boolToStr tab.italic},
+          strikethrough = ${boolToStr tab.strikethrough},
+          underline = "${tab.underline}",
+        }''; # No comma bc they're added in toWeztermColorscheme
+    in replaceStrings [ "\n" ] [''
+
+      ${indent}''] formatted;
+
+  toWeztermBase16Colors = colors:
+    let
+      values = with colors; [ black red green yellow blue magenta cyan white ];
+      joined = concatStringsSep ''", "'' values;
+    in ''{ "${joined}" }'';
+
+  toWeztermColorscheme = colors: indent:
+    let
+      formatted = ''
+        {
+          foreground = "${colors.foreground}",
+          background = "${colors.background}",
+          cursor_bg = "${colors.cursor.background}",
+          cursor_fg = "${colors.cursor.foreground}",
+          cursor_border = "${colors.cursor.border}",
+          selection_bg = "${colors.selection.background}",
+          selection_fg = "${colors.selection.foreground}",
+          scrollbar_thumb = "${colors.scrollbarThumb}",
+          split = "${colors.split}",
+          ansi = ${toWeztermBase16Colors colors.ansi},
+          brights = ${toWeztermBase16Colors colors.bright},
+          tab_bar = {
+            background = "${colors.tabBar.background}",
+            active_tab = ${
+              toWeztermTabColors colors.tabBar.activeTab "  ${indent}"
+            },
+            inactive_tab = ${
+              toWeztermTabColors colors.tabBar.inactiveTab "  ${indent}"
+            },
+            inactive_tab_hover = ${
+              toWeztermTabColors colors.tabBar.inactiveTabHover "  ${indent}"
+            },
+          },
+        },'';
+    in replaceStrings [ "\n" ] [''
+
+      ${indent}''] formatted;
```

Comment:
I'm unable to copy the link to the comment I sent to berbiche so I'll just copy it.

I agree it may be slightly unneeded, but majority of the people who I've talked to that want to use this module said they would prefer if it was formatted so they could read the generated code. If you feel like it should be removed then I can remove it.